falcon_kit/mains/ovlp_to_graph.py

```diff
@@ -1146,55 +1193,69 @@ def main(argv=sys.argv):
     # Currently, we use ad-hoc logic filtering out shorter utg, but we ca
     # add proper alignment comparison later to remove redundant utgs
 
-    utg_spurs = set()
     all_nodes = ug.nodes()
 
     ug2 = ug.copy()
-    spur_edges = set()
-    edges_to_remove = set()
 
-    for n in s_nodes:
-        if ug.in_degree(n) != 0:
+    s_candidates = set( )
+    for v in ug2.nodes():
+        if ug2.in_degree(v) == 0:
+            s_candidates.add(v)
+
+    while len(s_candidates) > 0:
```

Comment:
Actually,

    while s_candidates:

is sufficient. Whichever you prefer is fine.